project_stock/models/project_task.py

```diff
@@ -132,8 +152,10 @@ def _update_moves_info(self):
         self.action_assign()
 
     @api.model
-    def _prepare_procurement_group_vals(self, values):
-        return {"name": values["name"]}
+    def _prepare_procurement_group_vals(self):
+        """We use id in the name to avoid duplicate names if multiple tasks
+        have the same name."""
+        return {"name": "%s-%s" % (self.id, self.name)}
```

Comment:
The procurement group name should be short, but also unique. I recommend:

```suggestion
        return {"name": "Task-ID: %s" % self.id}
```